### PR TITLE
Add Templates

### PR DESCRIPTION
**Pull request by FarnazH**
1. Add minimal code and test
2. Setup Travis and other templates from `theochem/cookiecutter-python`

### CODE REVIEW

**Comment by FarnazH** [comment]
@tovrstra and @matt-chan, I am getting `could not download cache` error on Travis, can you please take a look?
I don't think it has to do with this, but I have not set up the GitHub, Anaconda, & PyPI tokens, because I don't have Roberto's credential to encrypt them.

**Comment by tovrstra** [comment]
The error regarding the caches is harmless. It is logical that no caches from previous travis builds are available because this is the first time Travis is running.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+- rm -rf $HOME/miniconda/locks $HOME/miniconda/pkgs $HOME/miniconda/var
+- pip uninstall -y cardboardlint # Cardboardlint always installs even if no changes are made.
+
+branches:
+  only:
+    - master
+    - /^[0-9]+\.[0-9]+(\.[0-9]+)?([ab][0-9]+)?$/
+
+before_install:
+#- if ! [[ $TRAVIS_TAG || $TRAVIS_OS_NAME == "linux" && $MYCONDAPY == "2.7" ]]; then exit 0; fi
```
Comment (tovrstra) [inline .travis.yml:101]: I'm not sure why this line is commented? I would suggest to uncomment it and to change "2.7" to "3.6".

Comment (tovrstra) [reply .travis.yml:101]: P.S. That may also be something to fix in the cookiecutter template?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,469 @@
+env:
+  matrix:
+#  - MYCONDAPY=2.7
+#  - MYCONDAPY=3.5
+  - MYCONDAPY=3.6
```
Comment (tovrstra) [inline .travis.yml:5]: Maybe add 3.7? (Useful for building release packages.)

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+- provider: pages
+  skip_cleanup: true
+  github_token: ${GITHUB_TOKEN}
+  project_name: ${PROJECT_NAME}
+  local_dir: doc/_build/html
+  on:
+    repo: ${GITHUB_REPO_NAME}
+    condition: "-e doc && $TRAVIS_TAG != *[ab]* && $MYCONDAPY == 2.7 && $TRAVIS_OS_NAME == linux"
+    tags: true
+####
```
Comment (tovrstra) [inline .travis.yml:262]: Everything below this line is overriding the settings above this line. I think the lower half show be removed? Can you check this is caused by the cookie-cutter or if it was due to another edit?

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+# ####
+# # EVERYTHING BELOW THIS LINE IS AUTOGENERATED.
+# ####
+
+# Do not use Travis Python to save some time.
+language: generic
+os:
+  - linux
+#  - osx
+#osx_image: xcode8.3
```
Comment (tovrstra) [inline .travis.yml:62]: Lines 61 and 62 can be uncommented (Useful for making conda packages for OSX.)

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+#  # not preparing a release. This is not possible on Travis. There are several tickets on
+#  # travis-ci related to this limitation.
+#  #
+#  #    https://github.com/travis-ci/travis-ci/issues/7451
+#  #    https://github.com/travis-ci/travis-ci/issues/7149
+#  #    https://github.com/travis-ci/travis-ci/issues/2778
+#  #    ...
+#  fast_finish: true
+#  allow_failures:
+#  - os: osx
```
Comment (tovrstra) [inline .travis.yml:82]: Lines 80, 81 and 82 can be uncommented. There is no need to wait for then when preparing a PR.

File: MANIFEST.in [hunk 6]
```diff
@@ -0,0 +1,4 @@
+include pylintrc
+include LICENSE.txt
+include clean.sh
```
Comment (tovrstra) [inline MANIFEST.in:3]: The file `clean.sh` is not present, so no need to include it in the MANIFEST.in file. I did not check the other files.

**Comment by tovrstra** [comment]
@FarnazH Can you check if the duplication in `.travis.yml` is due to a problem with the cookiecutter? Besides the minor points above, all seems to be good.

**Comment by kimt33** [comment]
We will stick with tox for now, but I may end up cherry-picking the commits for the readme and conda recipe.

**Pull request closed without merging**